nussl/core/effects.py

```diff
@@ -0,0 +1,584 @@
+    if min(delays) < 0 or max(delays) > 1000:
+        warnings.warn("One or more delays is far from the "
+                      "typical 40-60 ms range. This might produce strange results.", UserWarning)
+
+    if (len(delays) != len(decays) or len(decays)
+            != len(speeds) or len(speeds) != len(depths)):
+        raise ValueError("Delays, decays, depths, and speeds must all be the same length.")
+
+    delays = make_arglist_ffmpeg(delays)
+    speeds = make_arglist_ffmpeg(speeds)
+    decays = make_arglist_ffmpeg(decays)
+    depths = make_arglist_ffmpeg(depths)
+
+    return FFmpegFilter("chorus", in_gain=in_gain,
+                        out_gain=out_gain, delays=delays, speeds=speeds, decays=decays, depths=depths)
+
+
+def phaser(in_gain=.4, out_gain=.74, delay=3,
+           decay=.4, speed=.5, _type="triangular"):
+    """
+    https://ffmpeg.org/ffmpeg-all.html#aphaser
+
+    Creates a FFmpegFilter object, when called on an ffmpeg stream,
+    applies a phaser filter to the audio signal
+    Args:
+        in_gain (float): Proportion of input gain. Must be between 0 and 1
+        out_gain (float): Proportion of output gain. Must be between 0 and 1.
+        delay (float): Delay of chorus filter in ms. (Time between original signal and delayed)
+        decay (float): Decay of copied signal. Must be between 0 and 1.
+        speed (float): Modulation speed of the delayed filter. 
+        _type (str): modulation type. Either Triangular or Sinusoidal
+            "triangular" or "t" for Triangular
+            "sinusoidal" of "s" for sinusoidal
+    Returns:
+        filter (FFmpegFilter): Resulting filter, to be called on a ffmpeg stream
+    """
+    if in_gain > 1 or in_gain < 0:
+        raise ValueError("in_gain must be between 0 and 1.")
+    if out_gain > 1 or out_gain < 0:
+        raise ValueError("out_gain must be between 0 and 1.")
+    if decay > 1 or decay < 0:
+        raise ValueError("delay must be between 0 and 1.")
+
+    allowed_mod_types = {"triangular", "sinusoidal", "t", "s"}
+    if _type not in allowed_mod_types:
+        raise ValueError(f"_type must be one of the following:\n{allowed_mod_types}")
+
+    # type is reserved word in python, kwarg dict is necessary
+    type_kwarg = {
+        "type": _type
+    }
+
+    return FFmpegFilter("aphaser", in_gain=in_gain,
+                        out_gain=out_gain, delay=delay, speed=speed, decay=decay, **type_kwarg)
+
+
+def _flanger_argcheck(delay, depth, regen, width,
+                      speed, phase, shape, interp):
+    allowed_shape_types = {"triangular", "sinusoidal"}
+    if shape not in allowed_shape_types:
+        raise ValueError(f"shape must be one of the following:\n{allowed_shape_types}")
+    allowed_interp_types = {"linear", "quadratic"}
+    if interp not in allowed_interp_types:
+        raise ValueError(f"interp must be one of the following:\n{allowed_interp_types}")
+    if (delay < 0 or delay > 30
```

Comment:
Two things here:
1) I think you might be able to just do `if not (0 < delay < 30):`
2) I think it might be worth splitting each of these in its own if statement and throwing a separate exception for each. Yes it's ugly, but this whole section is already pretty hairy, so we might as well make things easier for the user by telling them exactly what's wrong.

Reply:
Changed to print only values that are out of bounds